### DISCUSSION TITLE
Two ideas to support code contribution

### DISCUSSION THREAD

**Discussion by RisingLeaf** [endless-sky/endless-sky#8090] [Ideas]
## Commenting:
I think I talked about this before with peter on one PR.
The idea is that we use a format for comments, that is parsable into a documentation, that way people could look up functions.
In general this would lower the barrier to be able to make code contributions.

## A Roadmap:
This one is something I would not only suggest for code, but for everything else. We currently have the development goals by MZ, but what we need in my opinion is a map of larger changes that will block development in one area until they are merged, that way people can see if it makes sense to develop a feature for this area right now. Cause it is definitely a waste of work to do something and then have to rewrite it, because there is some refractoring going on.

**Comment by RisingLeaf** [comment]
Addition, quick conversation with quyykk, result:
doxygen seems to be the right choice, as you can create a website from it and it is supported by various editors

**Comment by RisingLeaf** [comment]
Howdy, I actually generated a documentation for ES with doxygen: https://risingleaf.github.io/esdocs.github.io/

**Reply by RisingLeaf** [reply]
Look at this graph representing PlayerInfo :laughing: 
![image](https://github.com/endless-sky/endless-sky/assets/85687254/a5f2b1dd-8b55-4864-9f88-6e1098080450)

**Reply by RisingLeaf** [reply]
Seriously though I think it is a great resource because the documentation also includes comments that have been written above functions.

**Reply by Zitchas** [reply]
That's really neat!

**Comment by RisingLeaf** [comment]
Closing this as I opened a new discussion for the comment thing and the roadmap might see itself more clear in the vision documents, idk this discussion was never active anyways
